File: src/Blizka.App/Domain/Entities/DatePreference.cs
using Blizka.App.Domain.Enums;

namespace Blizka.App.Domain.Entities;

public sealed class DatePreference
{
    public Guid Id { get; set; }

    public DatePreferenceCode Code { get; set; }

    public string NameRu { get; set; } = string.Empty;

    public string NameBe { get; set; } = string.Empty;

    public string NameEn { get; set; } = string.Empty;

    public ICollection<UserDatePreference> UserDatePreferences { get; set; } = [];
}

File: frontend/src/domains/date-preferences/types/date-preference.ts
/** Сверено с backend: `Blizka.App/Domain/Enums/DatePreferenceCode.cs`. */
export type DatePreferenceCode =
  'activeOutdoors' | 'calmHangout' | 'quizzesBoardGames' | 'somethingNew'

export type DatePreference = {
  id: string
  /** Стабильный код, не зависит от локали — именно его принимает PATCH. */
  code: DatePreferenceCode
  name: string
}
